Yahoo source in v2 fails when Yahoo! cannot quote

Both `BTE` and `SPAQ` tickers returned no results (their Web site, however, does show prices).

I added the asserts below.  The error checking is not sufficient, it appears.  Sometimes `result` is flat out empty.

```
Traceback (most recent call last):
  File "/usr/bin/bean-price", line 11, in <module>
    load_entry_point('beancount==2.3.3', 'console_scripts', 'bean-price')()
  File "/usr/lib64/python3.8/site-packages/beancount/prices/price.py", line 650, in main
    price_entries = sorted(price_entries, key=lambda e: e.currency)
  File "/usr/lib64/python3.8/concurrent/futures/_base.py", line 611, in result_iterator
    yield fs.pop().result()
  File "/usr/lib64/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/lib64/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/usr/lib64/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/lib64/python3.8/site-packages/beancount/prices/price.py", line 439, in fetch_price
    srcprice = fetch_cached_price(source, psource.symbol, dprice.date)
  File "/usr/lib64/python3.8/site-packages/beancount/prices/price.py", line 347, in fetch_cached_price
    result = (source.get_latest_price(symbol)
  File "/usr/lib64/python3.8/site-packages/beancount/prices/sources/yahoo.py", line 90, in get_latest_price
    result = parse_response(response)
  File "/usr/lib64/python3.8/site-packages/beancount/prices/sources/yahoo.py", line 51, in parse_response
    assert 0, content
AssertionError: {'result': [], 'error': None}
```

```
Traceback (most recent call last):
  File "/usr/bin/bean-price", line 11, in <module>
    load_entry_point('beancount==2.3.3', 'console_scripts', 'bean-price')()
  File "/usr/lib64/python3.8/site-packages/beancount/prices/price.py", line 650, in main
    price_entries = sorted(price_entries, key=lambda e: e.currency)
  File "/usr/lib64/python3.8/concurrent/futures/_base.py", line 611, in result_iterator
    yield fs.pop().result()
  File "/usr/lib64/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/lib64/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/usr/lib64/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/lib64/python3.8/site-packages/beancount/prices/price.py", line 439, in fetch_price
    srcprice = fetch_cached_price(source, psource.symbol, dprice.date)
  File "/usr/lib64/python3.8/site-packages/beancount/prices/price.py", line 347, in fetch_cached_price
    result = (source.get_latest_price(symbol)
  File "/usr/lib64/python3.8/site-packages/beancount/prices/sources/yahoo.py", line 90, in get_latest_price
    assert 0, (response, ticker)
AssertionError: (<Response [200]>, 'BTE')
```

--- a/beanprice/sources/yahoo.py
+++ b/beanprice/sources/yahoo.py
@@ -44,6 +44,8 @@ def parse_response(response: requests.models.Response) -> Dict:
             ','.join(json.keys())))
     if content['error'] is not None:
         raise YahooError("Error fetching Yahoo data: {}".format(content['error']))
+    if not content['result']:
+        raise YahooError("No data returned from Yahoo, ensure that the symbol is correct")
     return content['result'][0]
 
 

--- a/beanprice/sources/yahoo_test.py
+++ b/beanprice/sources/yahoo_test.py
@@ -154,6 +154,12 @@ class YahooFinancePriceFetcher(unittest.TestCase):
         with self.assertRaises(yahoo.YahooError):
             yahoo.parse_response(response)
 
+    def test_parse_response_empty_result(self):
+        response = MockResponse(
+            '{"quoteResponse": {"error": null, "result": []}}')
+        with self.assertRaises(yahoo.YahooError):
+            yahoo.parse_response(response)
+
     def test_parse_response_no_timestamp(self):
         response = MockResponse(textwrap.dedent("""
             {"chart":

